testutils/file.go

```diff
@@ -30,12 +31,79 @@ func TempDir(dir, pattern string) (string, error) {
 	return dir, err
 }
 
-// TempFile returns a new unique temporary file using the given name.
-func TempFile(tb testing.TB, name string) *os.File {
+// TempFile creates a unique temporary file named "something.txt" or fails the test if it
+// cannot. It returns the file and a clean-up function.
+func TempFile(tb testing.TB) (*os.File, func()) {
 	tb.Helper()
-	tempFile := filepath.Join(tb.TempDir(), name)
+	tempFile := filepath.Join(tb.TempDir(), "something.txt")
 	//nolint:gosec
 	f, err := os.Create(tempFile)
 	test.That(tb, err, test.ShouldBeNil)
-	return f
+
+	return f, func() {
+		test.That(tb, f.Close(), test.ShouldBeNil)
+		// Since the file was placed in a directory that was created via TB.TempDir, it
+		// will automatically be deleted after the test and all its subtests complete, so
+		// we do not need to remove it manually.
+	}
+}
+
+// WatchedFiles creates a file watcher and n unique temporary files all named
+// "something.txt", or fails the test if it cannot. It returns the watcher, a slice of
+// files, and a clean-up function.
+//
+// For safety, this function will not create more than 50 files.
+func WatchedFiles(tb testing.TB, n int) (*fsnotify.Watcher, []*os.File, func()) {
+	tb.Helper()
+
+	if n > 50 {
+		tb.Fatal("will not create more than 50 temporary files, sorry")
+	}
```

Comment:
50 is arbitrary - can make it even smaller. just want to make sure no one accidentally a huge number of files by fat-fingering a few extra digits or something.